src/features/movies/MovieList/index.js

```diff
@@ -15,36 +15,51 @@ import {
   selectPopularMovies,
   selectPopularMoviesStatus,
   selectPopularMoviesTotalPages,
+  selectPopularMoviesTotalResults,
 } from "../popularMoviesSlice";
 import { MovieTile } from "../../../common/tiles/MovieTile";
 import { Loader } from "../../../common/states/Loader";
 import { Error } from "../../../common/states/Error";
 import { ReactComponent as Previous } from "./previousArrow.svg";
 import { ReactComponent as Next } from "./nextArrow.svg";
-
+import { Search } from "../../Search";
+import useQueryParameter from "../../Search/useQueryParameter";
+import { NoResult } from "../../../common/states/NoResult";
 
 export const MovieList = () => {
   const dispatch = useDispatch();
 
   const movies = useSelector(selectPopularMovies);
   const stateOfLoading = useSelector(selectPopularMoviesStatus);
+  const query = useQueryParameter("search");
 
   const [page, setPage] = useState(1);
   const totalPages = useSelector(selectPopularMoviesTotalPages);
+  const totalResults = useSelector(selectPopularMoviesTotalResults);
 
   useEffect(() => {
-    dispatch(fetchPopularMovies(page));
-  }, [dispatch, page]);
+    dispatch(fetchPopularMovies({ page, query }));
+  }, [dispatch, page, query]);
 
   return (
     <>
+      <Search />
       {stateOfLoading === "loading" ? (
-        <Loader title="Loading popular movies..." />
+        <Loader title="Loading..." />
       ) : stateOfLoading === "error" ? (
         <Error />
       ) : (
         <Wrapper>
-          <ListTitle> Popular movies</ListTitle>
+          {!movies.length ? (
+            <NoResult query={query} />
```

Comment:
W noResult wyswietla sie paginacja, takze trzeba bedzie dodac napewno jeszcze jeden warunek wyswietlania dotyczacy paginacji :)

Reply:
pewnie, zapisze sobie to gdzies